src/System.Net.Sockets/tests/FunctionalTests/OSSupport.cs

```diff
@@ -42,20 +43,120 @@ public void UseOnlyOverlappedIO_AlwaysFalse()
             }
         }
 
-        [PlatformSpecific(TestPlatforms.Windows)] // Windows IOCTL
         [Fact]
         public void IOControl_FIONREAD_Success()
         {
             using (var client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
             {
-                byte[] outValue = new byte[sizeof(int)];
+                Assert.Throws<SocketException>(() => client.IOControl(IOControlCode.DataToRead, null, null));
+                Assert.Throws<SocketException>(() => client.IOControl(IOControlCode.DataToRead, null, new byte[0]));
+                Assert.Throws<SocketException>(() => client.IOControl(IOControlCode.DataToRead, null, new byte[sizeof(int) - 1]));
 
-                const int FIONREAD = 0x4004667F;
-                Assert.Equal(4, client.IOControl(FIONREAD, null, outValue));
-                Assert.Equal(client.Available, BitConverter.ToInt32(outValue, 0));
+                byte[] siocatmarkResult = new byte[sizeof(int)];
```

Comment:
nit: rename this to fionreadResult/bytesAvailableResult?

Reply:
Yup. Will fix, thanks. The bigger issue I need to look into is why the SIOCATMARK test fails on macOS.

Reply:
I don't have a mac, so I'm pretty limited in debugging that. I do take an occasional peek in the mac kernel sources.
There is a comment on urgent pointers. Maybe something that is relevant here.
```
		 * According to RFC961 (Assigned Protocols),
		 * the urgent pointer points to the last octet
		 * of urgent data.  We continue, however,
		 * to consider it to indicate the first octet
		 * of data past the urgent section as the original
		 * spec states (in one of two places).
```

Reply:
Thanks, I opened #25639 to look into it.